### DISCUSSION TITLE
Compiling struggles :(

### DISCUSSION THREAD

**Discussion by davehouser1** [strongswan/strongswan#704] [Q&A] [answered]
Read through [this page ](https://wiki.strongswan.org/projects/strongswan/wiki/InstallationDocumentation)on how to compile. 
Also read though [this page ](https://wiki.strongswan.org/projects/strongswan/wiki/Autoconf)on available options.

I am unable to find a way to determine which specific options were enabled for a pre compiled swanctl binary. The binary came from a strongswan rpm install, which I believe is the same binary that comes from the public yum install repo.

I want to compile from source, but I don't know what options were used to create the original binary. The documentation reads _"note that many of these are enabled by default"_, what does this actually mean? Does it mean that the compiled binary in the rpm has these features already enabled? Or if I run `./configure` with nothing else, specific options will be enabled by default? Which ones?

My goal is to create a binary that will run with less privilege's, based off [this page ](note that many of these are enabled by default). The main issue I am running into is the problem with gmp not being recognized (even though its installed), I was able to work past it by using --disable-gmp and --enable-openssl. (Found from [this post](https://lists.strongswan.org/pipermail/dev/2013-February/000674.html)), but not the point of my post, the point is, did the binary from the rpm use these options as well? How can I tell?

Also, when I deploy with the rpm, it seems that path to the swanctl binary and systemd unit is deployed automatically. However doing a `make install` creates an odd service called strongswan-starter.service, and puts the swanctl binary in a location that is not apart of my path, but of which are different outcomes from the rpm install. I am well aware this is all because I am using the wrong options during the configure phase of compiling, however that is not the point of my post. This all stems back to, how do I list specific configuration options that were used for an original binary? What are these "default options" that are enabled by default? I want to create a binary exactly like the .rpm or yum repo creates but enable options for less privileges. 

Lastly, how do I uninstall a `make install` deployment? How can I tell where all files were put / remove them? is there a script that comes with the tar.bz2 to assist with this process?

**Comment by tobiasbrunner** [comment, marked as answer]
> I want to compile from source, but I don't know what options were used to create the original binary.

So maybe decide for yourself? But you should be able to see what options were enabled if you find the sources for that particular RPM (`.spec` file, see e.g. [here](https://src.fedoraproject.org/rpms/strongswan/tree/rawhide) for the Fedora sources). (I know very little about RPMs, though, so can't really help you all that much.)

> The documentation reads _"note that many of these are enabled by default"_, what does this actually mean?

Literally what it says. You see what the default values are for these options on the wiki page, or rather like in `configure --help` the options are actually listed as `--disable-...` as they are already enabled. You can also use `--disable-defaults` to disable all default options, you then have to enable all features manually (even stuff like `--enable-ikev2`).

> Does it mean that the compiled binary in the rpm has these features already enabled?

Probably some of them, but as mentioned above, the spec file will tell you what options were used.

> even though its installed

You might not have the header files for libgmp installed (`-devel` packages in the RPM world, `-dev` packages in the DEB world).

> I am well aware this is all because I am using the wrong options during the configure phase of compiling,

Correct, stuff like `--prefix` or `--sysconfdir` etc. are usually set to change the paths.

> Lastly, how do I uninstall a `make install` deployment?

Some files can be removed with `make uninstall`, but that's mostly binaries. Config files have to be removed manually (they are also not overwritten by later `make install` calls). If you haven't changed any paths via `configure`, these will be in `/usr/local/etc/` (other stuff will be in other directories in `/usr/local` as that's the default prefix).
